addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -447,14 +447,27 @@ async function loadImageInfo(img, rpc, attachmentSrc = '') {
         return;
     }
 
+    // Only consider the "relative" part of the URL. Needed because some
+    // relative URLs were wrongly converted to absolute URLs at some point and
+    // user domains could have been changed meanwhile.
+    let relativeSrc;
+    try {
+        const srcUrl = new URL(src);
+        relativeSrc = srcUrl.pathname;
+    } catch {
+        relativeSrc = src;
+    }
     const {original} = await rpc({
         route: '/web_editor/get_image_info',
-        params: {src: src.split(/[?#]/)[0]},
+        params: {src: relativeSrc.split(/[?#]/)[0]},
     });
-    // Check that url is local.
-    const isLocal = original && new URL(original.image_src, window.location.origin).origin === window.location.origin 
-        && !/\/web\/image\/\d+-redirect\//.test(original.image_src);
-    if (isLocal && original.image_src) {
+    // If src was an absolute "external" URL, we consider unlikely that its
+    // relative part matches something from the DB and even if it does, nothing
+    // bad happens, besides using this random image as the original when using
+    // the options, instead of having no option.
+    // The "redirect" check is for when it is a redirect image attachment due to
+    // an external URL upload.
+    if (original && original.image_src && !/\/web\/image\/\d+-redirect\//.test(original.image_src)) {
```

Comment:
@sobo-odoo In which case do we get the "redirect" url? I just tried when using the external URL upload, I got a base64 added in the DOM, then after a save a normal looking URL (with redirect), also there is an originalSrc + there does not seem to be an attachment with a redirect URL in my database :shrug:

Reply:
> If the image binary cannot be fetched by the client (e.g. CORS), the
old approach is used instead.

From what I understood in the commit message and by testing, it happens when the image cannot be fetched for cross-origins reasons
If the URL you are testing causes a "CORS error" (you see it in the console) 

 ![image](https://github.com/odoo/odoo/assets/102796037/19030d30-02e1-43a2-93a1-b9914346a294)

then a redirect image is created and it has no `data-original-*` attributes. It is the case with this random sky image URL for example (https://www.imagesdoc.com/wp-content/uploads/sites/33/2021/07/8536923965_5efb01fda3_o-2048x1322.jpg)  => it creates well a redirect

If you tested with `https://www.odoo.com/logo.png` then it works normally like you said because there is no CORS error :slightly_smiling_face: 

>  a normal looking URL (with redirect) 

Are you sure there was "redirect" in the URL ? (not doubting what you are saying but I don't have it so I wonder how you had it :thinking:)

Note that I compared the behavior with a master runbot and it seems to behave the same way

Reply:
>  If the URL you are testing causes a "CORS error" (you see it in the console)

Oh ok, thanks, I'll test with that :+1: 

> not doubting what you are saying but [...]

You should :smile: I meant to say "with**out** redirect". I'll test with the CORS error then :)